src/details/ArborX_Point.hpp

```diff
@@ -14,32 +14,51 @@
 
 #include <Kokkos_Macros.hpp>
 
+#include <array>
+
 namespace ArborX
 {
 class Point
 {
+private:
+  struct Data
+  {
+    double coords[3];
+  } _data;
+
 public:
   KOKKOS_INLINE_FUNCTION
-  constexpr double &operator[](unsigned int i) { return _coords[i]; }
+  constexpr Point()
+      : _data{}
+  {
+  }
```

Comment:
And my preference goes to a defaulted default constructor and an a default initializer when you declare the data member but at that point I am just nitpicking :)

Reply:
Couldn't get that to work in a `constexpr` context.

Reply:
Did you initialize `_data`?

Reply:
If I define the default `Point` constructor as `default`, the compiler complains that `coords` is not initialized and I can't define a constructor for `Data` without disallowing aggregate-initialization we use in the other constructor.

Reply:
https://godbolt.org/z/bqDcYB

Reply:
Yes, that works.